include/zephyr/net/net_if.h

```diff
@@ -2357,6 +2382,113 @@ void net_if_unset_promisc(struct net_if *iface);
  */
 bool net_if_is_promisc(struct net_if *iface);
 
+/**
+ * @brief Set implementation-specific interface options.
+ *
+ * @param iface Pointer to the network interface.
+ *
+ * @param optname Integer value representing the option to set.
+ *		  The meaning of values less than NET_IF_GLOB_OPT_BASE is up to the interface
+ *		  implementation.
+ *		  Values larger than or equal to NET_IF_GLOB_OPT_BASE are reserved for global
+ *		  iface options.
+ *
+ * @param optval Pointer to the value to be assigned to the option.
+ *
+ * @param optlen Length (in bytes) of the value to be assigned to the option.
+ *
+ * @return 0 if successful, -ENOTSUP if not implemented,
+ *	   implementation-specific error code otherwise.
+ */
+int net_if_set_opt(struct net_if *iface, int optname, const void *optval, size_t optlen);
+
+/**
+ * @brief Get implementation-specific interface options.
+ *
+ * @param iface Pointer to the network interface.
+ *
+ * @param optname Integer value representing the option to get.
+ *		  The meaning of values less than NET_IF_GLOB_OPT_BASE is up to the interface
+ *		  implementation.
+ *		  Values larger than or equal to NET_IF_GLOB_OPT_BASE are reserved for global
+ *		  iface options.
+ *
+ * @param optval Pointer to where the retrieved value should be stored.
+ *
+ * @param optlen Pointer to length (in bytes) of the destination buffer available for storing the
+ *		 retrieved value. If the available space is less than what is needed, or otherwise
+ *		 invalid, it is up to the interface implementation whether to:
+ *		 1) Truncate the retrieved value and return -ENOBUFS
+ *		 2) Refuse (with an error code) to retrieve the value at all
+ *
+ *		 If data is written to the optval pointer, optlen will be set to the total number
+ *		 of bytes written.
+ *
+ * @return 0 if successful, -ENOTSUP if not implemented, -ENOBUFS if not enough retrieval buffer
+ *	     provided, implementation-specific error code otherwise.
+ */
+int net_if_get_opt(struct net_if *iface, int optname, const void *optval, size_t *optlen);
+
+/**
+ * @brief Check whether a network interface has connectivity persistence enabled.
+ * Wraps a call to net_if_get_opt for NET_IF_OPT_CONNECTIVITY_PERSISTENT
+ *
+ * @param iface - Pointer to network interface to check.
+ * @return True if the option is enabled, otherwise False.
+ * Also returns false if the persistence value could not be retrieved or is not supported.
+ */
+bool net_if_is_connectivity_persistent(struct net_if *iface);
```

Comment:
Subjection but personally, I'd just drop these wrapper functions for the sake of API simplicity. If we see that some options are used very often then we can introduce them.

Reply:
I think I might agree if not for the fact that @rlubos has made a pretty convincing argument that maybe these options shouldn't be part of `net_if_get_opt` at all, in which case they cease to be helpers (and also become much simpler) -- What do you think of going in that direction instead?